### PR TITLE
Add Ironic tests

### PR DESCRIPTION
**Pull request by gabriel-samfira**
Adds some light weight ironic API tests.

### CODE REVIEW

File: zaza/openstack/charm_tests/ironic/setup.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    :param initrd_url: URL where the ari image resides
+    :type initrd_url: str
+    :param kernel_url: URL where the aki image resides
+    :type kernel_url: str
+    """
+    base_name = 'ironic-deploy'
+    initrd_name = "{}-initrd".format(base_name)
+    vmlinuz_name = "{}-vmlinuz".format(base_name)
+    if not initrd_url:
+        initrd_url = os.environ.get('FUNCTEST_IRONIC_DEPLOY_INITRD', None)
```
Comment (thedac) [inline zaza/openstack/charm_tests/ironic/setup.py:89]: Please make all variables begin with `TEST_`. We have been standardizing on variable names.

File: zaza/openstack/charm_tests/ironic/setup.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    if not kernel_url:
+        kernel_url = os.environ.get('FUNCTEST_IRONIC_DEPLOY_VMLINUZ', None)
```
Comment (thedac) [inline zaza/openstack/charm_tests/ironic/setup.py:91]: Please make all variables begin with `TEST_`. We have been standardizing on variable names.

File: zaza/openstack/charm_tests/ironic/setup.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+
+def add_ironic_os_image(image_url=None):
+    """Upload the operating system images built for bare metal deployments.
+
+    :param image_url: URL where the image resides
+    :type image_url: str
+    """
+    image_url = image_url or os.environ.get(
+        'FUNCTEST_IRONIC_RAW_BM_IMAGE', None)
```
Comment (thedac) [inline zaza/openstack/charm_tests/ironic/setup.py:115]: Please make all variables begin with `TEST_`. We have been standardizing on variable names.

File: zaza/openstack/charm_tests/glance/setup.py [hunk 4]
```diff
@@ -32,8 +32,36 @@ def basic_setup():
     """
 
 
+def _get_default_glance_client():
+    keystone_session = openstack_utils.get_overcloud_keystone_session()
```
Comment (thedac) [inline zaza/openstack/charm_tests/glance/setup.py:36]: Docstring

**Review by thedac — CHANGES_REQUESTED** [review]
Functionally, this looks great.  I have a couple of nitpicky requests.

1) Please change any environment variables so that they begin with `TEST_`
2) In the setup file please add any missing docstrings.

Thank you for the work.

**Comment by gabriel-samfira** [comment]
Done. PTAL.

Thanks for the review!

**Review by thedac — APPROVED** [review]
Looks good. Thanks.

**Pull request merged**